source/Coop.Core/Common/Services/Connection/Handlers/PlayerConnectionServerMessageHandler.cs

```diff
@@ -0,0 +1,76 @@
+﻿using Common.Logging;
+using Common.Messaging;
+using Common.Network;
+using Coop.Core.Client.Messages;
+using Coop.Core.Server.Connections;
+using Coop.Core.Server.Connections.Messages;
+using Coop.Core.Server.Services.Time.Messages;
+using GameInterface.Services.GameDebug.Messages;
+using LiteNetLib;
+using Serilog;
+using System.Linq;
+
+namespace Coop.Core.Common.Services.Connection.Handlers
+{
+    public class PlayerConnectionServerMessageHandler
+    {
+        private static readonly ILogger Logger = LogManager.GetLogger<PlayerConnectionServerMessageHandler>();
+
+        private readonly IMessageBroker messageBroker;
+        private readonly IClientRegistry clientRegistry;
+        private readonly INetwork network;
+
+        public PlayerConnectionServerMessageHandler(IMessageBroker messageBroker, IClientRegistry clientRegistry, INetwork network)
+        {
+            this.messageBroker = messageBroker;
+            this.clientRegistry = clientRegistry;
+            this.network = network;
+
+            messageBroker.Subscribe<NetworkPlayerCampaignEntered>(PlayerCampaignEnteredHandler);
+            messageBroker.Subscribe<PlayerConnected>(PlayerConnectedHandler);
+            messageBroker.Subscribe<NetworkRequestTimeSpeedChange>(NetworkRequestTimeSpeedChange);
+            messageBroker.Subscribe<NetworkConnected>(Handle_NetworkConnected);
+        }
+
+        internal void Handle_NetworkConnected(MessagePayload<NetworkConnected> obj)
+        {
+            messageBroker.Publish(this, new SendInformationMessage("A new player is joining the game, pausing"));
```

Comment:
The game does not pause exactly when a player connects, the game instead pauses [here](https://github.com/Bannerlord-Coop-Team/BannerlordCoop/blob/development/source/Coop.Core/Server/Connections/States/TransferSaveState.cs#L27)